### PR TITLE
Implement UpdateTicketModal

### PR DESCRIPTION
**Pull request by ryonryon**

### CODE REVIEW

**Comment by ryonryon** [comment]
https://user-images.githubusercontent.com/35892492/172981262-78bb1cb7-9d71-429c-992d-6fae1db63fcb.mov

File: src/components/TicketDetailModal/TicketDetailModal.tsx [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+                    />
+
+                    <DescriptionButtons>
+                      <DescriptionSaveButton title="Save" />
+                      <Button title="Cancel" />
+
+                      <Button title="Formatting help" />
+                    </DescriptionButtons>
+                  </DescriptionFormContainer>
+                </DescriptionForm>
```
Comment (ryonryon) [inline src/components/TicketDetailModal/TicketDetailModal.tsx:85]: I am going to replace this later.

File: src/components/Avatar/Avatar.test.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+      render(<Avatar alt={mockedAlt} size="S" />);
+
+      const headerClass = Avatar({ size: "S" }).type.styledComponentId;
+      const MyHeaderRoots = document.getElementsByClassName(headerClass);
+      const style = window.getComputedStyle(MyHeaderRoots[0]);
+
+      // Assert
+      expect(style.height).toBe("32px");
+      expect(style.width).toBe("32px");
```
Comment (ryonryon) [inline src/components/Avatar/Avatar.test.tsx:42]: @kei95 It is a bit tricky here. Let me know if you know better way to test.

File: src/components/Dialog/Dialog.test.tsx [hunk 3]
```diff
@@ -2,12 +2,12 @@ import Dialog from "./Dialog";
 import { fireEvent, render, screen } from "@testing-library/react";
 
 describe("<Dialog />", () => {
-  describe("Passed open", () => {
+  describe("Passed isOpen", () => {
```
Comment (ryonryon) [inline src/components/Dialog/Dialog.test.tsx:5]: I updated open to isOpen, I forgot to update it in last PR.

File: src/components/TicketDetailModal/TicketDetailModal.tsx [hunk 4]
```diff
@@ -0,0 +1,308 @@
+import {
```
Comment (ryonryon) [inline src/components/TicketDetailModal/TicketDetailModal.tsx:1]: This component is overwhelming. I populated with buttons. But only title, description and delete button will be functioning. So I only cared how it looks and it is not responsive UI.

File: src/components/Dialog/Dialog.tsx [hunk 5]
```diff
@@ -13,7 +13,7 @@ interface DialogProps {
 }
 
 export default function Dialog({
-  open,
+  isOpen,
   disableCloseButton = false,
```
Comment (kei95) [inline src/components/Dialog/Dialog.tsx:17]: [nitpick] - disableCloseButton could be `isDisableCloseButton`

File: src/components/TicketDetailModal/TicketDetailModal.tsx [hunk 6]
```diff
@@ -0,0 +1,308 @@
...
+    <Dialog isOpen={isOpen} onClose={onClose} disableCloseButton>
+      <Card>
+        <Container>
+          <CloseIconButton onClick={onClose}>
+            <Close />
+          </CloseIconButton>
+
+          <Headline>
+            <Title />
+
```
Comment (kei95) [inline src/components/TicketDetailModal/TicketDetailModal.tsx:58]: No offense, but when I see your JSX file I always feel the same as when I see teeth with gaps lol

image

File: src/components/TicketDetailModal/TicketDetailModal.tsx [hunk 7]
```diff
@@ -0,0 +1,308 @@
...
+            <Side>
+              <SideTitle>Add to card</SideTitle>
+              <Button title="Members" icon={<Person />} />
+              <Button title="Labels" icon={<Label />} />
+              <Button title="Checklist" icon={<Check />} />
+              <Button title="Dates" icon={<AccessTime />} />
+              <Button title="Attachment" icon={<AttachFile />} />
+              <Button title="Cover" icon={<Style />} />
+              <Button title="Custom Fields" icon={<CropLandscape />} />
```
Comment (kei95) [inline src/components/TicketDetailModal/TicketDetailModal.tsx:118]: You might be able to create an object that contains icon, title, and callback function(if you need) and iterate through with a map function. Doesn't have to be this way though

```typescript
// type for the object
type SideButton = { title: string, icon: JSX.Element(could be better type), onPress?: () => void};

// object to be mapped
const ADD_TO_CARD_BUTTONS: SideButton[] = [
  {title: "Members" icon: <Person />},
  {title:"Labels" icon: <Label />}
  ...
]

// JSX part 
<>
  {ADD_TO_CARD_BUTTONS.map((buttonItem) => 
      (<Button 
          title={buttonItem.title} 
          icon={buttonItem.icon} 
          onPress={buttonItem.onPress} />))
  };
</>
```

File: src/components/TicketDetailModal/TicketDetailModal.tsx (hunk 4 above)
Comment (kei95) [reply src/components/TicketDetailModal/TicketDetailModal.tsx:1]: I think you should split some parts of this component into sub-component. The name of this component is modal but considering the size, you can think of this as a kind of screen

File: src/components/TicketDetailModal/TicketDetailModal.tsx (hunk 1 above)
Comment (kei95) [reply src/components/TicketDetailModal/TicketDetailModal.tsx:85]: What are you going to replace this with?

**Review by kei95 — APPROVED** [review]

File: src/components/TicketDetailModal/TicketDetailModal.tsx (hunk 6 above)
Comment (ryonryon) [reply src/components/TicketDetailModal/TicketDetailModal.tsx:58]: HTML (or jsx) is not only tool to build a web page but a document. And shorter or small chunk is always easy to understand for other developers.

File: src/components/TicketDetailModal/TicketDetailModal.tsx (hunk 1 above)
Comment (ryonryon) [reply src/components/TicketDetailModal/TicketDetailModal.tsx:85]: Check how it is in Trello app

File: src/components/TicketDetailModal/TicketDetailModal.tsx (hunk 6 above)
Comment (kei95) [reply src/components/TicketDetailModal/TicketDetailModal.tsx:58]: My point is that they are too spacious - I think you are overdoing it.

File: src/components/TicketDetailModal/TicketDetailModal.tsx (hunk 4 above)
Comment (ryonryon) [reply src/components/TicketDetailModal/TicketDetailModal.tsx:1]: I would do that if I put all functionality for the buttons and everything.

Comment (kei95) [reply src/components/TicketDetailModal/TicketDetailModal.tsx:1]: Why not now? It's a small team that I don't bother but if you do it in a bigger team it's just making progress harder

Comment (ryonryon) [reply src/components/TicketDetailModal/TicketDetailModal.tsx:1]: My intention of implementing here is only delete ticket and update ticket. You can feel free to do this if you like it to do.

File: src/components/TicketDetailModal/TicketDetailModal.tsx (hunk 1 above)
Comment (kei95) [reply src/components/TicketDetailModal/TicketDetailModal.tsx:85]: What is wrong? I'm just asking you what you mean by your sentence.

**Pull request merged**